new/ohai-new-plugins-policy.md

```diff
@@ -0,0 +1,45 @@
+---
+RFC: unassigned
+Title: Ohai new plugins policy
+Author: Phil Dibowitz <phil@ipom.com>
+Status: Draft
+Type: <Standards Track, Informational, Process>
+---
+
+# Ohai new plugins policy
+
+To allow Chef to move fast while still providing stability, this lines out a
+policy for clarity.
+
+## Motivation
+
+    As a Chef developer,
+    I want to be able to provide new functionality to the community
+    so that they can use it.
+
+    As a Chef user,
+    I want to be able to leverage new functionality as quickly as possible
+    with minimal work while not breaking existing features.
+
+## Specification
+
+As of Ohai 14, most plugins are disabled by default. New plugins have no effect
```

Comment:
Only a few plugins should be disabled by default unless something major changed?

Reply:
I can go look at the code. I was under the impression we switched a whitelist by default.

Reply:
That said, even if they are run, again as long as they don't effect existing namespaces, I see no way in which this breaks compat.

Reply:
Ah, OK, so I misunderstood the `optional` feature, it's off by default not on. So **ideally** I'd want the policy to be:

* As long as you're not touching an existing namespace, it's fine to add

But I'd see at as reasonable to say:

* As long as you're not touching an existing nsmaespace, and it's marked optional, it's fine to add

Perhaps what the best compromise here is:

* In current, as long as you're not touching an existing namespace, it's fine to add
* In stable (current - 1), as long as your'e not touching an existing namespace and it's marked as optional, it's find to add